### PR TITLE
gh-101100: Fix Sphinx warnings in `library/cmd.rst`

### PR DESCRIPTION
**Pull request by hugovk**
Fix 13 Sphinx warnings:

* `do_foo`, `complete_foo`, `help_bar` and `do_bar`, `do_forward` and `do_playback` are examples, silence with `!`
* There's already a paragraph about `do_help`, let's add `.. method:: Cmd.do_help(arg)` so it can be linked to
* Silence `sys.stdout.write` and `sys.stdin.readline` with `!`. Or is there somewhere good to link to? 
* `cmdqueue` was missing the class ref, change to `~Cmd.cmdqueue`

```
Doc/library/cmd.rst:78: WARNING: py:meth reference target not found: do_foo
Doc/library/cmd.rst:78: WARNING: py:meth reference target not found: do_help
Doc/library/cmd.rst:78: WARNING: py:meth reference target not found: do_shell
Doc/library/cmd.rst:88: WARNING: py:meth reference target not found: complete_foo
Doc/library/cmd.rst:96: WARNING: py:meth reference target not found: do_help
Doc/library/cmd.rst:96: WARNING: py:meth reference target not found: help_bar
Doc/library/cmd.rst:96: WARNING: py:meth reference target not found: do_bar
Doc/library/cmd.rst:96: WARNING: py:meth reference target not found: do_help
Doc/library/cmd.rst:231: WARNING: py:meth reference target not found: sys.stdout.write
Doc/library/cmd.rst:231: WARNING: py:meth reference target not found: sys.stdin.readline
Doc/library/cmd.rst:251: WARNING: py:meth reference target not found: do_forward
Doc/library/cmd.rst:256: WARNING: py:meth reference target not found: do_playback
Doc/library/cmd.rst:256: WARNING: py:attr reference target not found: cmdqueue
```

* Issue: gh-101100

----
📚 Documentation preview 📚: https://cpython-previews--113502.org.readthedocs.build/

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]
LGTM. I have only doubts about silencing `sys.stdout.write` and `sys.stdin.readline`. Maybe add references to `sys.stdout` and `sys.stdin`?

**Comment by hugovk** [comment]
Sounds good, updated!

File: Doc/library/cmd.rst [hunk 1]
```diff
@@ -232,8 +232,8 @@ Instances of :class:`Cmd` subclasses have some public instance variables:
 .. attribute:: Cmd.use_rawinput
 
    A flag, defaulting to true.  If true, :meth:`cmdloop` uses :func:`input` to
-   display a prompt and read the next command; if false, :meth:`!sys.stdout.write`
-   and :meth:`!sys.stdin.readline` are used. (This means that by importing
+   display a prompt and read the next command; if false, :data:`sys.stdout.write <sys.stdout>`
+   and :data:`sys.stdin.readline <sys.stdin>` are used. (This means that by importing
```
Comment (serhiy-storchaka) [inline Doc/library/cmd.rst]:
```suggestion
   display a prompt and read the next command; if false, :data:`sys.stdout.write() <sys.stdout>`
   and :data:`sys.stdin.readline() <sys.stdin>` are used. (This means that by importing
```

File: Doc/library/cmd.rst [hunk 2]
```diff
@@ -76,27 +76,30 @@ A :class:`Cmd` instance has the following methods:
...
    arguments *text*, *line*, *begidx*, and *endidx*.  *text* is the string prefix
    we are attempting to match: all returned matches must begin with it. *line* is
    the current input line with leading whitespace removed, *begidx* and *endidx*
    are the beginning and ending indexes of the prefix text, which could be used to
    provide different completion depending upon which position the argument is in.
 
+
+.. method:: Cmd.do_help(arg)
+
    All subclasses of :class:`Cmd` inherit a predefined :meth:`do_help`.  This
```
Comment (AlexWaygood) [inline Doc/library/cmd.rst]:
```suggestion
   All subclasses of :class:`Cmd` inherit a predefined :meth:`!do_help`.  This
```

File: Doc/library/cmd.rst [hunk 3]
```diff
@@ -76,27 +76,30 @@ A :class:`Cmd` instance has the following methods:
...
    method, called with an argument ``'bar'``, invokes the corresponding method
-   :meth:`help_bar`, and if that is not present, prints the docstring of
-   :meth:`do_bar`, if available.  With no argument, :meth:`do_help` lists all
+   :meth:`!help_bar`, and if that is not present, prints the docstring of
+   :meth:`!do_bar`, if available.  With no argument, :meth:`do_help` lists all
```
Comment (AlexWaygood) [inline Doc/library/cmd.rst]:
```suggestion
   :meth:`!do_bar`, if available.  With no argument, :meth:`!do_help` lists all
```

**Review by AlexWaygood — COMMENTED** [review]
Nitpick: no need to link to the thing we're already reading (otherwise, this looks great!)

**Review by AlexWaygood — APPROVED** [review]
LGTM!

**Pull request merged**